isisd/isis_cli.c

```diff
@@ -332,6 +332,39 @@ void cli_show_ip_isis_ipv6(struct vty *vty, struct lyd_node *dnode,
 		yang_dnode_get_string(dnode, "../area-tag"));
 }
 
+/*
+ * XPath: /frr-interface:lib/interface/frr-isisd:isis/bfd-monitoring
+ */
+DEFPY(isis_bfd,
+      isis_bfd_cmd,
+      "[no] isis bfd",
+      NO_STR
+      PROTO_HELP
+      "Enable BFD support\n")
+{
+	const struct lyd_node *dnode;
+
+	dnode = yang_dnode_get(vty->candidate_config->dnode,
+			       "%s/frr-isisd:isis", VTY_CURR_XPATH);
+	if (dnode == NULL)
+		return CMD_SUCCESS;
+
+	nb_cli_enqueue_change(vty, "./frr-isisd:isis/bfd-monitoring",
+			      NB_OP_MODIFY, no ? "false" : "true");
+
+	return nb_cli_apply_changes(vty, NULL);
+}
+
+void cli_show_ip_isis_bfd_monitoring(struct vty *vty, struct lyd_node *dnode,
+				     bool show_defaults)
+{
+	if (show_defaults == false)
+		vty_out(vty, " %s" PROTO_NAME " bfd\n",
+			yang_dnode_get_bool(dnode, NULL) ? "" : "no ");
+	else
+		vty_out(vty, " no " PROTO_NAME " bfd\n");
```

Comment:
For boolean leafs we can use the following idiom:
```
        if (!yang_dnode_get_bool(dnode, NULL))
                vty_out(vty, " no");

        vty_out(vty, PROTO_NAME " bfd\n");
```

This way we don't need to update this callback if we change the default value from `false` to `true` in the YANG module (not that this will happen :smile:).

Also, your callback always displays `no isis bfd` when `with-defaults` is given, which is wrong.